src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidBrokerFactory.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Android.App;
+using Android.Content;
+using Microsoft.Identity.Client.ApiConfig.Parameters;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Internal;
+using Microsoft.Identity.Client.Internal.Broker;
+using Microsoft.Identity.Client.Internal.Requests;
+using Microsoft.Identity.Client.OAuth2;
+using Microsoft.Identity.Client.UI;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client.Platforms.Android.Broker
+{
+    internal class AndroidBrokerFactory
+    {
+        private static BrokerType s_installedBroker;
+
+        private enum BrokerType 
+        {
+            None,
```

Comment:
Not really "None", but "Unknown"

Reply:
None since this will be the case when no broker is installed. But then initially it will be Unknown.

Reply:
`NoneOrUnknown` ?

Reply:
Or code both for clarity.